fbw-common/src/systems/instruments/src/PFD/AttitudeIndicatorWarnings.tsx

```diff
@@ -0,0 +1,237 @@
+import { DisplayComponent, FSComponent, MappedSubject, Subject, VNode } from '@microsoft/msfs-sdk';
+
+import { Arinc429RegisterSubject, Arinc429Word, ArincEventBus } from '@flybywiresim/fbw-sdk';
+import { RopRowOansSimVars, FwcDataEvents, TawsDataEvents } from '../MsfsAvionicsCommon/providers';
+
+interface AttitudeIndicatorWarningsProps {
+  bus: ArincEventBus;
+  instrument: BaseInstrument;
+}
+
+export class AttitudeIndicatorWarnings extends DisplayComponent<AttitudeIndicatorWarningsProps> {
+  private readonly warningGroupRef = FSComponent.createRef<SVGGElement>();
+
+  private readonly maxReverseActive = Subject.create(false);
+
+  private readonly maxReverseMaxBrakingActive = Subject.create(false);
+
+  private readonly ifWetRwyTooShortActive = Subject.create(false);
+
+  private readonly rwyTooShortActive = Subject.create(false);
+
+  private readonly rwyAheadActive = Subject.create(false);
+
+  private readonly fwcWord126 = Arinc429RegisterSubject.createEmpty();
+
+  private readonly gpwsWord2 = Arinc429RegisterSubject.createEmpty();
+
+  // stall warning = stall bit && !on ground bit
+  private readonly stallActive = this.fwcWord126.map((v) => v.bitValueOr(17, false) && !v.bitValue(28));
+
+  // FIXME no source yet
+  private readonly stopRudderInputActive = Subject.create(false);
+
+  private readonly windshearActive = this.gpwsWord2.map((w) => w.bitValueOr(15, false));
+
+  // FIXME no source yet
+  private readonly wsAheadCaution = Subject.create(false);
+
+  // FIXME no source yet
+  private readonly wsAheadWarning = Subject.create(false);
+
+  onAfterRender(node: VNode): void {
+    super.onAfterRender(node);
+
+    const sub = this.props.bus.getSubscriber<FwcDataEvents & RopRowOansSimVars & TawsDataEvents>();
+
+    sub
+      .on('rowRopWord1Raw')
+      .whenChanged()
+      .handle((raw) => {
+        const ar = new Arinc429Word(raw);
```

Comment:
I think those won't change much once implemented, but as a note: I think it would be better to move those to Arinc429Register as well to avoid the allocations and GC. Not critical here now but just in general 👍

Reply:
Good point, thanks! I changed them to Arinc429RegisterSubjects